examples/contrib/cifar10/main.py

```diff
@@ -75,6 +75,50 @@ def training(local_rank, config):
     # Create trainer for current task
     trainer = create_trainer(model, optimizer, criterion, lr_scheduler, train_loader.sampler, config, logger)
 
+    # Finding optimal Learning Rate using FastaiLRFinder
+    if config["use_lr_finder"]:
+        lr_finder = FastaiLRFinder()
+        to_save = {"model": model, "optimizer": optimizer}
+        with lr_finder.attach(
+            trainer, to_save, output_transform=lambda output: output["batch loss"], start_lr=0.2
+        ) as trainer_with_lr_finder:
+            trainer_with_lr_finder.run(train_loader)
+        logger.info(f"\tOptimal Learning Rate: {lr_finder.lr_suggestion()}")
+        # Remove the attached lr_scheduler to create a new one
+        trainer.remove_event_handler(lr_scheduler, Events.ITERATION_STARTED)
+        # if config["with_clearml"]:
+        #     lr_finder_plot = lr_finder.plot()
+        #     task.logger.report_matplotlib_figure(
+        #         title="LRFinder plot", series="LR vs Loss", figure=lr_finder_plot
+        #     )
+        # Create new scheduler with the new optimal learning rate
+        config["learning_rate"] = lr_finder.lr_suggestion()
+        le = config["num_iters_per_epoch"]
+        milestones_values = [
+            (0, 0.0),
+            (le * config["num_warmup_epochs"], config["learning_rate"]),
+            (le * config["num_epochs"], 0.0),
+        ]
+        lr_scheduler = PiecewiseLinear(optimizer, param_name="lr", milestones_values=milestones_values)
```

Comment:
Instead of redefining `lr_scheduler` instance, can't we rewrite just its `milestones_values` ?

Reply:
Yes that can be done, thanks for the recommendation!

Reply:
Sorry, I tried this but it won't work that way, because of this
https://github.com/pytorch/ignite/blob/master/ignite/handlers/param_scheduler.py#L932

Reply:
You can just update middle values of `lr_scheduler.values` and put a comment explaining what you are doing.

Reply:
That's done, thanks for the suggestions!